luatest/server.lua

```diff
@@ -106,21 +181,53 @@ function Server:initialize()
     end
 end
 
---- Generates environment to run process with.
--- The result is merged into os.environ().
--- @return map
+--- Create a table with env variables based on the constructor params.
+-- The result will be passed into the server process.
+-- Table consists of the following entries:
+--
+--   * TARANTOOL_WORKDIR
+--   * TARANTOOL_LISTEN
+--   * TARANTOOL_ALIAS
+--   * TARANTOOL_HTTP_PORT
+--   * TARANTOOL_BOX_CFG
+--
+-- @return table
 function Server:build_env()
-    return {
+    local res = {
         TARANTOOL_WORKDIR = self.workdir,
-        TARANTOOL_HTTP_PORT = self.http_port,
         TARANTOOL_LISTEN = self.net_box_port or self.net_box_uri,
         TARANTOOL_ALIAS = self.alias,
     }
+    if self.http_port ~= nil then
+        res.TARANTOOL_HTTP_PORT = self.http_port
```

Comment:
And if `self.http_port` is `nil`, then `res.TARANTOOL_HTTP_PORT` is also `nil`, right? So anyway it equals `self.http_port`.

Reply:
Fixed.